app/src/main/java/com/example/tasky/agenda/presentation/itemdetail/viewmodel/EventDetailViewModel.kt

```diff
@@ -68,6 +68,19 @@ class EventDetailViewModel @Inject constructor(
         }
     }
 
+    fun deletePhoto(savedStateHandle: SavedStateHandle) {
```

Comment:
Not a fan of having this here
Since you are adding logic of photo deletion, on the navigation

This should be on the Event screen (eventviewmodel)

Reply:
You could access the imageurl on the eventviewmodel with savedstatehandle, and remove the photo from there

Reply:
the problem is that the savedStateHandle injected in the viewModel doesnt get the changes of the PhotoViewerScreen. Im not sure which is the bes way to handle "Navigation back with result" the video from pihllip on YT also uses the entry that the composable function provides